src/hotspot/cpu/aarch64/templateInterpreterGenerator_aarch64.cpp

```diff
@@ -201,7 +201,7 @@ address TemplateInterpreterGenerator::generate_math_entry(AbstractInterpreter::M
     continuation = r19;
     __ ldrd(v0, Address(esp, 2 * Interpreter::stackElementSize));
     __ ldrd(v1, Address(esp));
-    __ mov(sp, r13);
+    __ mov(sp, r19_sender_sp);
```

Comment:
Ok, finally a real bug! At line 200 we have
```
 200     __ mov(r19, lr);   <== this destroys sp in r19_sender_sp
 201     continuation = r19;
```

So, when you switch this from using `r13` to using `r19` here at line 204 you are effectively updating `sp` with `lr`.

Note this doesn't happen in the previous cases at line 193 because the assignments were done in the opposite order.